tasks/lib/generateHTMLReport.js

```diff
@@ -33,16 +33,26 @@ module.exports = function generateHTMLReport(errorFileObj, options, errorFileCou
             dateFormat;
 
         timePortion = timePortion.substr(0, timePortion.lastIndexOf(":")).replace(/:/g, "-");
-        
+
         dateFormat = datePortion + "-" + timePortion;
         folderPath = (options.useTimeStamp === true) ? "w3cErrors-"+ dateFormat : "w3cErrors";
     }
-    var filePathTemp = curruntErrorFile["filename"].split("/"),
-        filePath;
 
-    filePathTemp = (filePathTemp[filePathTemp.length-1].indexOf(".") === -1) ? filePathTemp.slice(filePathTemp.length-2).join("") : filePathTemp.slice(filePathTemp.length-2).join("").split(".")[0];
+    var filePath;
 
-    filePath = filePathTemp + "_validation-report" + ".html";
+    if (!options.errorFileName) {
+        var filePathTemp = curruntErrorFile["filename"].split("/");
+
+        filePathTemp = (filePathTemp[filePathTemp.length-1].indexOf(".") === -1) ? filePathTemp.slice(filePathTemp.length-2).join("") : filePathTemp.slice(filePathTemp.length-2).join("").split(".")[0];
+
+        filePathTemp.replace(/\,\<\>\?\|\*\:\"/, '');
+
+        filePath = filePathTemp + "_validation-report" + ".html";
+    } else if (typeof options.errorFileName === 'string') {
+        filePath = options.errorFileName;
```

Comment:
Hi Marius,

filePath = options.errorFileName; line will overwrite the error file with same name because if user is testing for multiple URLs/files then it will always have the last tested file data. I believe the function name is a better option so please update your code and send me another Pull request.